client/src/Models/Learn/Filter.model.ts

```diff
@@ -0,0 +1,201 @@
+import _ from "lodash";
+import { Catalog } from "./Catalog.model";
+import { FilterType } from "./FilterType.model";
+import { LearnContent } from "./LearnContent.model";
+import { Product } from "./Product.model";
+import { observable, action } from "mobx";
+import { getRegexs, scoreRegex } from '../../Features/MicrosoftLearn/MicrosoftLearnFilterUtils'
+
+export class Filter {
+    @observable displayFilters: Map<FilterType, string[]>;
```

Comment:
I was trying to create a separate model class for `Filter` to segregate the logic and keep the `LearnStore` cleaner by having a `Filter` object and operate on the same from there. The observables in the class were added as otherwise the changes were not being reflected to the internal members properly. Read [this solution](https://stackoverflow.com/questions/47227740/mobx-doesnt-observe-classes-in-observable-array) and it seemed to work. Although a separate store for the Filter as per your other comment would be much cleaner and I thought the same initially as well but wasn't sure how I would implement the dependency between the two stores therein. I will do some research on the same.

Reply:
The newer implementation implements `Filter` as a separate store which almost answers my query. Hence, resolving the conversation.